### ISSUE TITLE
Remove deprecated twisted.python.dispatch

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5023]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#5023|
|Type|task|
|Created|2011-04-04 19:31:20Z|
|Branch|https://github.com/twisted/twisted/tree/python-dispatch-5023|

[twisted/python/dispatch.py](twisted/python/dispatch.py) was deprecated 5 years ago and should be removed.

<details><summary>Searchable metadata</summary>

```
trac-id__5023 5023
type__task task
reporter__thijs thijs
priority__low low
milestone__ 
branch__branches_python_dispatch_5023 branches/python-dispatch-5023
branch_author__thijs thijs
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1301945480000000 1301945480000000
changetime__1302193451000000 1302193451000000
version__None None
owner__thijs thijs
cc__thijs cc__spiv
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_1"></a>|@thijstriemstra commented|
|-|-|

(In [31551]) Branching to 'python-dispatch-5023'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [31552]) remove module, add NEWS file. refs [#5023](https://github.com/twisted/twisted/issues/5023)

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|spiv commented|
|-|-|

Buildbot results, when they are ready: http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/python-dispatch-5023

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_5"></a>|spiv commented|
|-|-|

This looks completely fine to me.  It's hard to argue with removing an unused module that's been deprecated since 2006!  Thanks for removing cruft and making Twisted smaller :)

Please merge.

**Comment by twisted-trac** [comment]
|<img alt="spiv's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|spiv set owner to @thijstriemstra|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_7"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [31567]) Removed the unused twisted.python.dispatch module, deprecated in 2006.

Author: thijs
Reviewer: spiv
Fixes: [#5023](https://github.com/twisted/twisted/issues/5023)
